vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -744,6 +750,18 @@ public void removeThemeVariants(DatePickerVariant... variants) {
                         .collect(Collectors.toList()));
     }
 
+    /**
+     * Gets the feature flags for the current UI.
+     * <p>
+     * Exposed with protected visibility to support mocking
+     *
+     * @return the current set of feature flags
+     */
+    protected FeatureFlags getFeatureFlags() {
+        return FeatureFlags
+                .get(UI.getCurrent().getSession().getService().getContext());
```

Comment:
```suggestion
                .get(VaadinService.getCurrent().getContext());
```

Even tho I have a feeling that this could create a problem / bottle neck if e.g. the context isn't available and it called like literally every-time a field is instantiated, which could be hundred times on a single view. Should this value be cached somewhere and not queried every time again? r.h. `FieldFeatureFlagUtils::isEnforceFieldValidationEnabled` as static method somewhere which can cache the value in a thread local if you wanna "over-engineer" it, or just a simple static boolean

Reply:
Looking at the implementation of `FeatureFlags.get`, it seems to cache feature flags to an attribute of the Vaadin context object, so getting these flags here should be quite cheap. Do you rather mean that calling `VaadinService.getCurrent().getContext()` can be not efficient?

Reply:
I'm thinking more about the call to `VaadinService.getCurrent().getContext()` but I could be totally wrong

Reply:
I'm also hesitent to use `UI::getCurrent` or other `getCurrent` implementations like above, because I've seen issue with it in the past e.g. exceptions inside the `DatePicker` constructor (where I can't find the issue)

Reply:
`VaadinService.getCurrent()` would return an existing instance from a Map data structure, so it shouldn't be a big deal, and then `getContext()` is always returning (not exactly a classic singleton instance, but still) a single existing instance of `VaadinContext`, so it shouldn’t be an expensive operation either.

Reply:
Thanks for the update - this looks better :) This comment can be closed